docs/advanced-recipes/atom-creators.mdx

```diff
@@ -291,3 +291,61 @@ function EvenCounter() {
   return <>Count was set to an even number {evenCount} times.</>
 }
 ```
+
+## atomWithBroadcast
+
+> `atomWithBroadcast` creates an atom. The atom will be shared between
+> browser tabs and frames, similar to `atomWithStorage` but with the 
+> initialization limitation.
+
+This can be useful when you want the state to interact with each other without
+the use of localStorage and that by just using The Broadcast Channel API allows
+basic communication between browsing contexts (that is, windows, tabs, frames, 
+create a component or iframes) and workers on the same origin.
+
+```ts
+import { atom } from "jotai";
+
+export function atomWithBroadcast<Value>(key: string, initialValue: Value) {
+  const channel = new BroadcastChannel(key);
+
+
+  const baseAtom = atom(initialValue);
+
+  const broadcastAtom = atom<Value, { fromOnMessage: boolean; value: Value }>(
+    (get) => get(baseAtom),
+    (get, set, update) => {
+      if (!update.fromOnMessage) {
+        channel.postMessage(update.value);
+      }
+      set(baseAtom, update.value);
+    }
+  );
+  broadcastAtom.onMount = (setAtom) => {
+    channel.onmessage = (event) => {
```

Comment:
Isn't this overwriting event listener if we use multiple providers?

Reply:
on second thought, I don't think this needs to be in `onMount`. Can you move it right after L312?